nbconvert/preprocessors/execute.py

```diff
@@ -494,14 +517,14 @@ def _get_timeout(self, cell):
 
         return timeout
 
-    def _handle_timeout(self):
+    def _handle_timeout(self, timeout, cell=None):
         self.log.error(
-            "Timeout waiting for execute reply (%is)." % self.timeout)
+            "Timeout waiting for execute reply (%is)." % timeout)
```

Comment:
This was because the specified `timeout` is not always `self.timeout`, i.e. in the case of the IOPub timeout.